### PR TITLE
Added new Biomarkers schema. Planning to add script validations but h…

### PR DESCRIPTION
**Pull request by hknahal**
…ave empty 'script' fields as placeholders for now.
Related to https://github.com/icgc-argo/argo-dictionary/issues/278

### CODE REVIEW

File: schemas/biomarkers.json [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+        "primaryId": true,
+        "displayName": "Submitter Treatment ID"
+      },
+      "restrictions": {
+        "regex": "#/regex/submitter_id",
+        "script": ""
+      }
+    },
+    {
+      "description": "If the biomarker test was done during a follow up event, then indicate the associated submitter_follow_up_id here.",
```
Comment (rosibaj) [inline schemas/biomarkers.json:79]: `follow-up` to be consistent with the current dictionary terminology

File: schemas/biomarkers.json [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+        "primaryId": true,
+        "displayName": "Submitter Follow-Up ID"
+      },
+      "restrictions": {
+        "regex": "#/regex/submitter_id",
+        "script": ""
+      }
+    },
+    {
+      "description": "If the biomarker test was not associated with a specific followup, primary diagnosis or treatment event, then indicate the interval of time since primary diagnosis that the biomarker test was performced at, in days.",
```
Comment (rosibaj) [inline schemas/biomarkers.json:93]: ` follow-up` here as well.

File: schemas/biomarkers.json [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+        "range": {
+           "exclusiveMin": 0
+        }
+      },
+      "meta": {
+        "displayName": "CA19-9 Level"
+      }
+    },
+    {
+      "description": "Indicate the quantitative measurement of the amount of CRP, an inflammatory marker, in the blood in mg/L. (Reference: NCIt C64548)",
```
Comment (rosibaj) [inline schemas/biomarkers.json:124]: should CRP be spelled out `...  amount of  C-reactive protein (CRP)...`?

File: schemas/biomarkers.json [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+        "range": {
+           "exclusiveMin": 0
+        }
+      },
+      "meta": {
+        "displayName": "C-reactive protein (CRP) Level"
+      }
+    },
+    {
+      "description": "Indicate the level of lactate dehydrogenase, in IU/L.",
```
Comment (rosibaj) [inline schemas/biomarkers.json:137]: `...level of lactate dehydrogenase (LDH)..`

File: schemas/biomarkers.json [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+        "range": {
+           "exclusiveMin": 0
+        }
+      },
+      "meta": {
+        "displayName": "Lactate Dehydrogenase (LDH) Level"
+      }
+    },
+    {
+      "description": "Indicate the value for a hemotology labratory test for the absolute number of neutrophil cells present in a sample of peripheral blood from a donor, in cells/uL. (Reference: caDSR CDE ID 2180198)",
```
Comment (rosibaj) [inline schemas/biomarkers.json:150]: labratory --> `laboratory`?

File: schemas/biomarkers.json [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+           "Positive",
+           "Unknown"
+        ]
+      },
+      "meta": {
+        "displayName": "Estrogen Receptor Status"
+      }
+    },
+    {
+      "description": "Indicate the Allred score. The Allred score is based on the percentage of cells that stain positive by IHC for estrogen receptor (ER) and the intensity of that staining. (Reference: NAACCR: 3828, caDSR CDE ID 2725288)",
```
Comment (rosibaj) [inline schemas/biomarkers.json:211]: for consistency with below:
` ... that stain positive by immunohistochemistry (IHC)...`

File: schemas/biomarkers.json [hunk 7]
```diff
@@ -0,0 +1,481 @@
...
+      "name": "pr_status",
+      "valueType": "string",
+      "restrictions": {
+        "codelist": [
+           "Cannot be determined",
+           "Negative",
+           "Not applicable",
+           "Positive",
+           "Unknown"
+
```
Comment (rosibaj) [inline schemas/biomarkers.json:281]: missing bracket here

File: schemas/biomarkers.json [hunk 8]
```diff
@@ -0,0 +1,481 @@
...
+           "Unknown"
+        ],
+        "script": ""
+      },
+      "meta": {
+        "displayName": "Progesterone Receptor Allred Score"
+      }
+    },
+    {
+      "description": "Indicate the immunohistochemical test result that refers to the overexpression or lack of expression of PD-L1 (programmed death ligand 1) in a tissue sample of a primary or metastatic malignant neoplasm. (Reference NCIt: C122807)",
```
Comment (rosibaj) [inline schemas/biomarkers.json:311]: to be consistent with above: 
`... programmed death ligand 1 (PD-L1)...`

File: schemas/biomarkers.json [hunk 9]
```diff
@@ -0,0 +1,481 @@
...
+           "Not applicable",
+           "Unknown"
+        ]
+      },
+      "meta": {
+        "displayName": "Pan-TRK IHC Status"
+      }
+    },
+    {
+      "description": "Indicate the expression of gene arrangement involving the RET proto-oncogene as assessed by fluoresence in situ hybridization (FISH). RET gene rearrangements are associated with several different neoplastic conditions. (Reference: NCIt C46005)",
```
Comment (rosibaj) [inline schemas/biomarkers.json:413]: should this be `RET proto-oncogene (RET1) ` ?

File: schemas/biomarkers.json [hunk 10]
```diff
@@ -0,0 +1,481 @@
...
+      "meta": {
+        "displayName": "ROS1 IHC Status"
+      }
+    },
+    {
+      "description": "Indicate the expression of Pan-TRK as assessed by immunohistochemistry (IHC).",
+      "name": "pan-trk_ihc_status",
+      "valueType": "string",
+      "restrictions": {
+        "codelist": [
```
Comment (rosibaj) [inline schemas/biomarkers.json:400]: genral comment - should this be one common codelist?

**Review by rosibaj — CHANGES_REQUESTED** [review]

File: schemas/biomarkers.json (hunk 1 above)
Comment (hknahal) [reply schemas/biomarkers.json]: Updated to `follow-up` to be consistent

File: schemas/biomarkers.json (hunk 2 above)
Comment (hknahal) [reply schemas/biomarkers.json]: done

File: schemas/biomarkers.json (hunk 3 above)
Comment (hknahal) [reply schemas/biomarkers.json]: Updated to `C-reactive protein (CRP)`

File: schemas/biomarkers.json (hunk 4 above)
Comment (hknahal) [reply schemas/biomarkers.json]: Updated to include `'(LDH)`

File: schemas/biomarkers.json (hunk 5 above)
Comment (hknahal) [reply schemas/biomarkers.json]: corrected

File: schemas/biomarkers.json (hunk 6 above)
Comment (hknahal) [reply schemas/biomarkers.json]: Updated to include this

File: schemas/biomarkers.json (hunk 10 above)
Comment (hknahal) [reply schemas/biomarkers.json]: Good point Rosi. I aggregated this into a common codelist (`expression_status`)

**Review by rosibaj — APPROVED** [review]

**Comment by hknahal** [comment]
Added validation script to biomarker schema. At least one of [`submitter_specimen_id`, `submitter_primary_diagnosis_id`, `submitter_treatment_id`, `submitter_follow_up_id`] OR `test_interval` is required. If more than one is submitted for a given row, an error will occur.


Tested in QA using version `108.3`:

- If `submitter_specimen_id`, `submitter_primary_diagnosis_id`, `submitter_treatment_id` or `submitter_follow_up_id` is not submitted, then `test_interval` is required.
- If `test_interval` is submitted, then `submitter_specimen_id`, `submitter_primary_diagnosis_id`, `submitter_treatment_id` or `submitter_follow_up_id` should not be submitted.

**Review by rosibaj — APPROVED** [review]
Tested on QA with the mentioned dictionary - was not able to violate any of the script rules.

**Review by rosibaj — APPROVED** [review]

**Pull request merged**